src/Logger/Adapters/DailyAdapter.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Quantum PHP Framework
+ *
+ * An open source software development framework for PHP
+ *
+ * @package Quantum
+ * @author Arman Ag. <arman.ag@softberg.org>
+ * @copyright Copyright (c) 2018 Softberg LLC (https://softberg.org)
+ * @link http://quantum.softberg.org/
+ * @since 2.9.5
+ */
+
+namespace Quantum\Logger\Adapters;
+
+use Quantum\Libraries\Storage\FileSystem;
+use Quantum\Logger\ReportableInterface;
+use Quantum\Exceptions\LangException;
+use Quantum\Exceptions\DiException;
+use Quantum\Logger\LoggerException;
+use ReflectionException;
+use Quantum\Di\Di;
+
+/**
+ * Class DailyAdapter
+ * @package Quantum\Logger
+ */
+class DailyAdapter implements ReportableInterface
+{
+
+    /**
+     * @var FileSystem
+     */
+    private $fs;
+
+    /**
+     * @var string
+     */
+    private $logFile;
+
+    /**
+     * DailyAdapter constructor.
+     * @param array $params
+     * @throws DiException
+     * @throws LoggerException
+     * @throws ReflectionException
+     * @throws LangException
+     */
+    public function __construct(array $params)
+    {
+        $this->fs = Di::get(FileSystem::class);
+
+        if(!$this->fs->isDirectory( $params['path'])) {
+            throw LoggerException::logPathIsNotDirectory($params['path']);   
+        }
+        
+        $this->logFile = $params['path'] . DS . date('Y-m-d') . '.log';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function report(string $level, $message, ?array $context = [])
```

Comment:
Why don't we create a ParentAdapter for repetitive functions (report and formatMessage)